src/System.Private.ServiceModel/tests/Common/Infrastructure/xunit/WcfTestCase.cs

```diff
@@ -44,13 +50,48 @@ internal WcfTestCase(IXunitTestCase testCase, string skippedReason = null, bool
 
         public void Deserialize(IXunitSerializationInfo info) { _testCase.Deserialize(info); }
 
-        public Task<RunSummary> RunAsync(
+        public async Task<RunSummary> RunAsync(
             IMessageSink diagnosticMessageSink, IMessageBus messageBus, object[] constructorArguments,
             ExceptionAggregator aggregator, CancellationTokenSource cancellationTokenSource)
         {
-            return _isTheory
-                ? new XunitTheoryTestCaseRunner(this, DisplayName, _skippedReason, constructorArguments, _diagnosticMessageSink, messageBus, aggregator, cancellationTokenSource).RunAsync()
-                : new XunitTestCaseRunner(this, DisplayName, _skippedReason, constructorArguments, TestMethodArguments, messageBus, aggregator, cancellationTokenSource).RunAsync();
+            ConcurrentQueue<EventWrittenEventArgs> events = new ConcurrentQueue<EventWrittenEventArgs>();
+	          s_testListener._eventWritten = events.Enqueue;
+
+            RunSummary runsummary = await (_isTheory ? new XunitTheoryTestCaseRunner(this, DisplayName, _skippedReason, constructorArguments, _diagnosticMessageSink, messageBus, aggregator, cancellationTokenSource).RunAsync()
+                : new XunitTestCaseRunner(this, DisplayName, _skippedReason, constructorArguments, TestMethodArguments, messageBus, aggregator, cancellationTokenSource).RunAsync());
+
+            if (runsummary.Failed > 0)
+            {
+                s_testListener._eventWritten = null;
+                StringBuilder etwOutput = new StringBuilder();
+                etwOutput.AppendLine(string.Format("---ETW Trace for Test {0} Begins---", DisplayName));
+                foreach (var item in events)
+                {
+                    try
+                    {
+                        etwOutput.AppendLine(string.Format(DisplayName + ":" + item.Message, item.Payload));
+                    }
+                    //The mumber of parameters in Payload does not match the number of arguments in the item.Message and thus cause a 
+                    // FormatException occationally, In this case, we catch and output all items in the payload and the Message without formatting the message.
+                    // https://github.com/dotnet/wcf/issues/1440 is opened to investigate the root cause of the mismatch exception.
+                    catch (Exception)
+                    {
+                        etwOutput.AppendLine(string.Format("ETW message: {0}, payload below was received", item.Message));
```

Comment:
>etwOutput.AppendLine(string.Format("ETW message: {0}, payload below was received", item.Message)); [](start = 24, length = 98)

Since we are trying to understand issue #1440, I recommend we write to the StringBuilder something to help us debug, something like this:
etwOutput.AppendLine(String.Format("ETW message encountered exception '{0}' using DisplayName '{1}', format '{2}' and {3} payload items", 
                                            exception.Message, DisplayName, item.Message, item.Payload.Count()));